src/lib/FreeC/LiftedIR/Syntax/Type.hs

```diff
@@ -44,3 +44,23 @@ data Type
       , typeFreeArg :: Type
       }
  deriving (Eq, Show)
+
+-------------------------------------------------------------------------------
+-- Utility                                                                   --
+-------------------------------------------------------------------------------
+
+-- | Applies a lifted IR type constructor to a type level expression.
+typeApp :: Type -> Type -> Type
+typeApp (TypeCon srcSpan name ts dec) t = TypeCon srcSpan name (t : ts) dec
+typeApp _ _ = error "Cannot apply type to non type constructor!"
+
+-- | Smart constructor for lifted IR functions.
+func :: Type -> Type -> Type
+func = FuncType NoSrcSpan
```

Comment:
I don't like smart constructors that just drop the `SrcSpan` argument. They encourage the user to drop location information. In the IR there is for example the following smart constructor which does exactly what you use `LIR.func` for, but does force the user to think about relevant location information (or actively decide that there is no `SrcSpan` than can be inserted).

```haskell
-- | Creates a function type with the given argument and return types.
funcType :: SrcSpan -> [Type] -> Type -> Type
funcType srcSpan = flip (foldr (FuncType srcSpan))
```

Also note that smart constructor names should match the constructor names.